packages/manager/src/managers/prismicRepository/PrismicRepositoryManager.ts

```diff
@@ -482,44 +475,20 @@ export class PrismicRepositoryManager extends BaseManager {
 			);
 
 			if (error) {
-				return {
-					error: new UnexpectedDataError(
-						`Failed to decode environments: ${error.errors.join(", ")}`,
-					),
-				};
+				throw new UnexpectedDataError(
+					`Failed to decode environments: ${error.errors.join(", ")}`,
+				);
 			}
 
 			if ("results" in value) {
 				let environments = value.results;
 
-				// Only include the user's personal environment
-				// by default. We must filter in the manager
-				// because the API returns all personal
-				// environments.
-				if (!args?.includeAllPersonalEnvironments) {
-					try {
-						const profile = await this.user.getProfile();
-
-						environments = environments.filter((environment) => {
-							if (environment.kind === "dev") {
-								return environment.users.some(
-									(user) => user.id === profile.shortId,
-								);
-							}
-
-							return true;
-						});
-					} catch (e) {
-						if (isUnauthenticatedError(error)) {
-							return { error };
-						}
+				if (!args?.includeAll) {
+					const profile = await this.user.getProfile();
 
-						return {
-							error: new UnexpectedDataError(
-								"Unexpected Error while fetching Environments",
-							),
-						};
-					}
+					environments = environments.filter((environment) =>
+						environment.users.some((user) => user.id === profile.shortId),
```

Comment:
So even non dev envs contain the authorized user shortId?

Reply:
Correct, the `users` array contains all users with access to the environment. Confirmed with the `prismic-main` and `environments-test` repositories.